client/rest-high-level/src/test/java/org/elasticsearch/client/CrudIT.java

```diff
@@ -624,6 +629,69 @@ public void testBulk() throws IOException {
         validateBulkResponses(nbItems, errors, bulkResponse, bulkRequest);
     }
 
+    public void testReindex() throws IOException, InterruptedException {
+        final String sourceIndex = "source1";
+        final String destinationIndex = "dest";
+        {
+            // Prepare
+            Settings settings = Settings.builder()
+                .put("number_of_shards", 1)
+                .put("number_of_replicas", 0)
+                .build();
+            createIndex(sourceIndex, settings);
+            createIndex(destinationIndex, settings);
+            assertEquals(
+                RestStatus.OK,
+                highLevelClient().bulk(
+                    new BulkRequest()
+                        .add(new IndexRequest(sourceIndex, "type", "1")
+                            .source(Collections.singletonMap("foo", "bar"), XContentType.JSON))
+                        .add(new IndexRequest(sourceIndex, "type", "2")
+                            .source(Collections.singletonMap("foo2", "bar2"), XContentType.JSON)),
+                    RequestOptions.DEFAULT
+                ).status()
+            );
+        }
+        {
+            TimeUnit.SECONDS.sleep(1);
```

Comment:
@nik9000 I know this is incorrect. But I couldn't make the test pass without it. Will firing a refresh request help?